routes/shopItemById.js

```diff
@@ -1,38 +1,33 @@
 const express = require("express");
-shopItemById = express.Router({ mergeParams: true });
-const utility = require("../utility.js");
-const db = require("../server.js");
+const sql = require("../database/database.js");
 
-shopItemById.get("/", (req, res, next) => {
+const shopItemById = express.Router({ mergeParams: true });
 
-});
-
-shopItemById.post("/", (req, res, next) => {
-
-});
+const result = async (id) => {
+    const data = await sql`
+    DELETE FROM grocery_list AS g WHERE g.grocery_id = ${id}
+    `
+    return data;
+};
 
 shopItemById.delete("/", (req, res, next) => {
-    try {
-        req.params.id = Number(req.params.id);
-        if (!isNaN(req.params.id)) {
-            db.run("DELETE FROM grocery_list WHERE id = $id", {
-                $id: req.params.id
-            }, function(err) {
-                if (err) {
-                    console.log(err);
-                    res.status(500).send();
-                } else if (this.changes === 0) {
+    req.params.id = Number(req.params.id);
+    if (!isNaN(req.params.id)) {
+        result(req.params.id)
+            .then(() => {
+                if (!req.params.id) {
```

Comment:
What this supposed to test

Reply:
I've changed these lines to the one below: 
`try {
            const data = await result(req.params.id);
            if (data.count === 0) {
                res.status(404).send();
            } else {
                res.status(204).send();
            }
        }
        catch (error) {
            console.log(error);
            res.status(500).send();
        }
`

Basically the data.count property is the number of affected rows returned by the database. So I can check in each put or delete request if anything has been changed.